src/Microsoft.Framework.Configuration/ConfigurationBase.cs

```diff
@@ -69,16 +71,14 @@ public IList<IConfigurationSource> Sources
 
         public IEnumerable<IConfigurationSection> GetChildren()
         {
-            var prefix = GetPrefix();
-
             var segments = Sources.Aggregate(
                 Enumerable.Empty<string>(),
-                (seed, source) => source.ProduceConfigurationSections(seed, prefix, Constants.KeyDelimiter));
+                (seed, source) => source.ProduceConfigurationSections(seed, GetPath(), Constants.KeyDelimiter));
 
             var distinctSegments = segments.Distinct();
             return distinctSegments.Select(segment =>
             {
-                return new ConfigurationSection(Sources, prefix + segment);
+                return new ConfigurationSection(Sources, GetPath(), segment);
```

Comment:
But what I meant is that we should just really pass the parentPath here (without the delimiter), and the concatenation of the delimiter should happen (if necessary) in the constructor of ConfigurationSection.